src/vs/workbench/browser/parts/paneCompositePart.ts

```diff
@@ -39,6 +40,33 @@ import { getActionBarActions } from '../../../platform/actions/browser/menuEntry
 import { IHoverService } from '../../../platform/hover/browser/hover.js';
 import { HiddenItemStrategy, MenuWorkbenchToolBar } from '../../../platform/actions/browser/toolbar.js';
 import { DeferredPromise } from '../../../base/common/async.js';
+import { IConfigurationService } from '../../../platform/configuration/common/configuration.js';
+import { URI } from '../../../base/common/uri.js';
+
+/**
+ * Context for chat view title actions.
+ * Defined locally to avoid layering violations with contrib layer.
+ */
+interface IChatViewTitleActionContext {
+	$mid: MarshalledId.ChatViewContext;
+	sessionResource: URI;
+}
+
+/**
+ * Internal interface for accessing chat view widget properties.
+ * Defined locally because these are internal implementation details not exposed via public interfaces,
+ * and importing from contrib layer would violate layering rules.
+ */
+interface IChatViewWithWidget {
+	widget?: {
+		viewModel?: {
+			model?: {
+				sessionResource?: URI;
+				getRequests?: () => unknown[];
+			};
+		};
+	};
+}
```

Comment:
@benibenj @bpasero wasn't quite sure what to do here so for the moment I've defined local interfaces.

Was specifically unsure about making some of the private properties for model resources/requests public, so again just defined an interface locally to make TS happy. Would love input from @roblourens or @justschen on if this feels icky or not.

Reply:
I'm not an expert in this area, but I don't think there should be anything chat-specific in AbstractPaneCompositePart. Rather we should figure out how to turn it into a generic feature, which the chat view can enable/configure in some way, or, we figure out how to have a chat-specific `IPaneCompositePart` which has chat features. We might ask Steven for advice, or maybe Ben understands it well.